cirq-google/cirq_google/devices/named_topologies_test.py

```diff
@@ -0,0 +1,106 @@
+# Copyright 2021 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import itertools
+from typing import Iterable
+from unittest.mock import MagicMock
+
+import cirq
+import networkx as nx
+import pytest
+from cirq_google import (
+    draw_gridlike,
+    LineTopology,
+    TiltedSquareLattice,
+    get_placements,
+    draw_placements,
+)
+import cirq_google as cg
+
+
+def test_tilted_square_lattice():
+    width = 2
+    height = 3
```

Comment:
There are corner cases here due to the fact that parity of each dimension leads to slightly different codepaths. I suggest testing a bunch of sizes. The extra test could for example check that the number of edges is always height times weight:

```python
@pytest.mark.parameterize('width, height', [
    (1, 1), (2, 1), (1, 2), (2, 2), (3, 5), (4, 4),
])
def test_tilted_square_lattice_sizes(width, height):
    topo = TiltedSquareLattice(width, height)
    assert len(topo.graph.edges) == width * height
```

If you want more aggressive tests, you could also verify the number of nodes of each degree or check connectedness and planarity

```python
assert nx.is_connected(topo.graph)
assert nx.algorithms.planarity.check_planarity(topo.graph)
```

Such high-level checks are easily tripped by bugs, so I think they're good tests. They also read well.

Reply:
Thanks; yeah this test needs to be beefed up. 

Thanks for the suggestions! TIL the number of edges = width * height. Do you know a simple(r) formula for the number of nodes?

In any event, I parameterized the test, kept the original checks, and added all the checks you suggested.

Reply:
The `edges = width * height` relationship becomes apparent when we remove all vertices and replace edges with dots.

The formula for the number of vertices is not that nice, but you can derive it by summing big and small Xes in the asciiart in the docstring. There are `(width//2 + 1) * (height//2 + 1)` small xes and `((width + 1)//2) * ((height + 1)//2)` big ones, so I think it should be

```python
(width//2 + 1) * (height//2 + 1) + ((width + 1)//2) * ((height + 1)//2)
```

It's nicer if you put the lattice on a torus...

Reply:
Awesome, added https://github.com/quantumlib/Cirq/pull/4370/commits/502b0aa2eb09253c8e80eb85cbcc5b978972c721